astrowidgets/glupyter.py

```diff
@@ -0,0 +1,400 @@
+
+        viewer = self._viewer
+        data = validate_data_argument(self._app.data_collection, data)
+
+        # NOTE: Changing state via API does not work properly if we loaded
+        # multiple data into the same widget in the same session.
+        self._app.data_collection.clear()
+
+        self._app.add_datasets(data)
+        viewer.add_data(data)
+        viewer.state.reference_data = data  # Bring it to focus
+        self._link_image_to_cb()
+
+    def load_array(self, arr, label=None):
+        """Load a 2D array into the viewer.
+
+        .. note:: Use :meth:`load_nddata` for WCS support.
+
+        Parameters
+        ----------
+        arr : array-like
+            2D array.
+
+        label : str or `None`
+            Glue label name for the data.
+
+        """
+        if label is None:
+            label = _gen_random_label()
+
+        data = Data(label=label)
+        data.add_component(component=arr, label=label)
+
+        viewer = self._viewer
+        data = validate_data_argument(self._app.data_collection, data)
+
+        # NOTE: Changing state via API does not work properly if we loaded
+        # multiple data into the same widget in the same session.
+        self._app.data_collection.clear()
+
+        self._app.add_datasets(data)
+        viewer.add_data(data)
+        viewer.state.reference_data = data  # Bring it to focus
+        self._link_image_to_cb()
+
+    def center_on(self, point):
+        raise NotImplementedError  # FIXME
+
+    def offset_to(self, dx, dy, skycoord_offset=False):
+        raise NotImplementedError  # FIXME
+
+    @property
+    def zoom_level(self):
+        raise NotImplementedError  # FIXME
+
+    @zoom_level.setter
+    def zoom_level(self, value):
+        raise NotImplementedError  # FIXME
+
+    # Need this here because we need to overwrite the setter.
+    @property
+    def marker(self):
+        return super().marker
+
+    @marker.setter
+    def marker(self, value):
+        raise NotImplementedError  # FIXME
+
+    def get_markers_by_name(self, marker_name, x_colname='x', y_colname='y',
+                            skycoord_colname='coord'):
+        raise NotImplementedError  # FIXME
+
+    def add_markers(self, table, x_colname='x', y_colname='y',
+                    skycoord_colname='coord', use_skycoord=False,
+                    marker_name=None):
+        raise NotImplementedError  # FIXME
+
+    def remove_markers_by_name(self, marker_name):
+        raise NotImplementedError  # FIXME
+
+    @property
+    def stretch_options(self):
+        # Options extracted from Glue
+        return ['arcsinh', 'linear', 'log', 'sqrt']
+
+    @property
+    def stretch(self):
+        return self._viewer.state.layers[0].stretch
+
+    @stretch.setter
+    def stretch(self, value):
+        self._viewer.state.layers[0].stretch = value
+
+    @property
+    def autocut_options(self):
+        raise NotImplementedError  # FIXME: Does Glue support this?
```

Comment:
I don't understand what this does but you can set the percentile values in glue to automatically determine the levels/cuts in the image viewer. Could you clarify what this should return?

Reply:
For Ginga, this is what happens:

```python
>>> w.autocut_options
('minmax', 'median', 'histogram', 'stddev', 'zscale')
>>> w.cuts = 'zscale'  # Automatically sets the cut levels according to this algorithm
```

But if this does not make sense for Glue (or other future backends), then maybe I'll take `autocut_options` out of the abstract class and make it a Ginga-only option... 💭